### PR TITLE
S3 endpoint setup

### PR DESCRIPTION
**Pull request by tiagograf-AOT**
*Issue #:*
https://github.com/bcgov/met-public/issues/22

*Description of changes:*
Integration with S3 storage
Upload images in engagement form.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: met-api/src/met_api/models/engagement.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ class Engagement(db.Model):
     published_date = db.Column(db.DateTime, nullable=True)
     content = db.Column(db.Text, unique=False, nullable=False)
     rich_content = db.Column(JSON, unique=False, nullable=False)
-    banner_url = db.Column(db.String(), unique=False, nullable=True)
+    banner_filename = db.Column(db.String(), unique=False, nullable=True)
```
Comment (saravanpa-aot) [inline met-api/src/met_api/models/engagement.py:32]: in the scriipt , its banner_url

File: met-api/src/met_api/services/object_storage_service.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    def get_auth_headers(self, documents: List[Document]):
+        """Get the s3 auth headers or the provided documents."""
+        if(_Config.OSS_S3_ACCESS_KEY_ID is None or
+            _Config.OSS_S3_SECRET_ACCESS_KEY is None or
+            _Config.OSS_S3_HOST is None or
+            _Config.OSS_S3_BUCKET is None
+           ):
+            return {'status': 'Configuration Issue',
+                    'message': 'accesskey is None or secretkey is None or S3 host is None or formsbucket is None'}, 500
```
Comment (saravanpa-aot) [inline met-api/src/met_api/services/object_storage_service.py:29]: ideally throw an exception here since this is configuration missing ;but ok for now

**Review by saravanpa-aot — APPROVED** [review]

File: met-api/migrations/versions/d0f92ae9ba77_alter_engagement_alter_banner_url.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+revision = 'd0f92ae9ba77'
+down_revision = '2545d45bb29c'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('engagement', sa.Column('banner_filename', sa.String(), nullable=True))
+    op.drop_column('engagement', 'banner_url')
```
[the remark this answers is not in the document]
Comment (tiagograf-AOT) [reply met-api/migrations/versions/d0f92ae9ba77_alter_engagement_alter_banner_url.py:22]: we store banner_filename, banner_url is gone from db but used in the frontend to display the image

File: met-api/src/met_api/config.py [hunk 4]
```diff
@@ -76,6 +76,13 @@ class _Config():  # pylint: disable=too-few-public-methods
     JWT_OIDC_CACHING_ENABLED = os.getenv('JWT_OIDC_CACHING_ENABLED', 'True')
     JWT_OIDC_JWKS_CACHE_TIMEOUT = 300
 
+    OSS_S3_BUCKET = os.getenv('OSS_S3_BUCKET')
```
[the remark this answers is not in the document]
Comment (tiagograf-AOT) [reply met-api/src/met_api/config.py]: removed

File: met-api/src/met_api/services/object_storage_service.py (hunk 2 above)
Comment (tiagograf-AOT) [reply met-api/src/met_api/services/object_storage_service.py:29]: noted

File: met-web/src/components/engagement/form/ActionContext.tsx [hunk 5]
```diff
@@ -114,8 +135,21 @@ export const ActionProvider = ({ children }: { children: JSX.Element }) => {
         );
     };
 
-    const handleUpdateEngagementRequest = (engagement: EngagementForm) => {
+    const handleUploadBannerImage = async () => {
+        if (!bannerImage) {
+            return savedBannerImageFileName;
+        }
```
[the remark this answers is not in the document]
Comment (tiagograf-AOT) [reply met-web/src/components/engagement/form/ActionContext.tsx:141]: noted

**Pull request merged**